src/frontends/tensorflow_common/src/op/cumsum.cpp

```diff
@@ -20,6 +26,24 @@ OutputVector translate_cumsum_op(const NodeContext& node) {
     auto exclusive = node.get_attribute<bool>("exclusive", false);
     auto reverse = node.get_attribute<bool>("reverse", false);
 
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(x.get_node_shared_ptr());
+    if (complex_type_mark) {
+        x = complex_type_mark->input_value(0);
+        auto zero = create_same_type_const_scalar<int32_t>(x, 0);
+        
+        auto is_zero = make_shared<v1::Equal>(axis, zero);
+        auto greater_than_zero = make_shared<v1::Greater>(axis, zero);
+
+        auto logical_or = make_shared<v1::LogicalOr>(is_zero, greater_than_zero);
+
+        auto const_one = make_shared<v0::Constant>(element::i32, Shape{}, 1);
+        auto const_minus_one = make_shared<v0::Constant>(element::i32, Shape{}, -1);
+
+        auto axis_update = make_shared<v1::Select>(logical_or, const_one, const_minus_one);
+
+        auto new_axis = make_shared<v1::Add>(axis, axis_update);    
```

Comment:
I think update for `axis` should be different. If `axis>=0` update is not needed, that mean update equal to const_zero because auxiliary dim (for concatenation of real and imag parts) do not affect cumsum,
otherwise, (for negative axis) update should be `rank-1`, where rank is a rank of `x = complex_type_mark->input_value(0);`. In order to compute rank, please use `ShapeOf` operations twice, see other examples

Reply:
Right, I  was under the assumption that the complex type values axis would be added in-front of the tensor rather than the back. Clarified it now, raise a revision.